### PR TITLE
Migrating Random Forest tests from boost to catch2

### PR DESCRIPTION
**Pull request by Abilityguy**
Regarding issue #2523

### CODE REVIEW

File: src/mlpack/tests/random_forest_test.cpp [hunk 1]
```diff
@@ -324,19 +321,19 @@ BOOST_AUTO_TEST_CASE(LeafSizeDatasetTest)
...
+    REQUIRE(predictions[i] == majorityClass);
     for (size_t j = 0; j < probabilities.n_rows; ++j)
-      BOOST_REQUIRE_CLOSE(probabilities(j, i), majorityProbs[j], 1e-5);
+      REQUIRE(probabilities(j, i) == Approx(majorityProbs[j]).epsilon(1e-7));
   }
 }
 
 // Make sure we can serialize a random forest.
-BOOST_AUTO_TEST_CASE(SerializationTest)
+TEST_CASE("RandomForestSerializationTest", "[RandomForestTest]")
```
Comment (Abilityguy) [inline src/mlpack/tests/random_forest_test.cpp:336]: Here, I slightly changed the name of the test case. When I retained the name 'SerializationTest', I got the following error 

`error: TEST_CASE( "SerializationTest" ) already defined. First seen at mlpack/mlpack/src/mlpack/tests/random_forest_test.cpp:337 Redefined at mlpack/mlpack/src/mlpack/tests/sparse_coding_test.cpp:140`

So I changed the name to 'RandomForestSerializationTest'. Kindly let me know if this is ok.

Comment (zoq) [reply src/mlpack/tests/random_forest_test.cpp:336]: Yes, that's completely fine.

**Review by zoq — APPROVED** [review]
Looks great to me, the failing Windows build is unrelated with the changes from the PR.

**Review by birm — APPROVED** [review]
Thanks for taking the time to do this!

**Comment by Abilityguy** [comment]
You are welcome! :)
Thanks for the quick reviews @zoq and @birm.

**Pull request merged**